src/Symfony/Component/VarDumper/Cloner/AbstractCloner.php

```diff
@@ -336,17 +350,48 @@ protected function castObject(Stub $stub, bool $isNested): array
             $parents[] = '*';
 
             $r = new \ReflectionClass($class);
+            $classAttributes = $r->getAttributes(AbstractAttribute::class, \ReflectionAttribute::IS_INSTANCEOF);
+
+            $properties = [];
+            foreach ($r->getProperties() as $property) {
+                $properties[$property->name] = [
+                    'value' => $property->getValue($obj),
+                    'visibility' => $property->isPublic() ? Visibility::Public : ($property->isProtected() ? Visibility::Protected : Visibility::Private),
+                ];
+            }
+
             $fileInfo = $r->isInternal() || $r->isSubclassOf(Stub::class) ? [] : [
                 'file' => $r->getFileName(),
                 'line' => $r->getStartLine(),
             ];
 
-            $this->classInfo[$class] = [$i, $parents, $hasDebugInfo, $fileInfo];
+            $this->classInfo[$class] = [$i, $parents, $hasDebugInfo, $fileInfo, $classAttributes, $properties];
         }
 
         $stub->attr += $fileInfo;
         $a = Caster::castObject($obj, $class, $hasDebugInfo, $stub->class);
 
+        foreach ($classAttributes as $attribute) {
+            $attribute = $attribute->newInstance();
+
+            if ($attribute instanceof SensitiveElement && !($this->flags & self::CLONER_WITH_SENSITIVE_ELEMENTS)) {
```

Comment:
this whole concern should be dealt with in a caster to me - in Caster::castObject here
we'd need to find a way to pass the attributes to them